website/project/decorators.py

```diff
@@ -193,6 +212,10 @@ def check_can_access(node, user, key=None, api_node=None):
     if user is None:
         return False
     if not node.can_view(Auth(user=user)) and api_node != node:
+        if request.args.get('action', '') == 'download':
+            if check_can_download_preprint_file(user, node):
```

Comment:
Reasonably certain this block is unnecessary -- see question 1 above. Also, when writing code of the form 
```python
if function_that_returns_a_boolean(*args): 
    return True
```
the preferred syntax is 
```python
return function_that_returns_a_boolean(*args)
```

Reply:
well, I thought about that but since this line would be hit in many download actions (some not preprint related), `check_can_download_preprint_file` returning `False` would prevent the exceptions below from being raised or  node request access form not be displayed.